types/core.go

```diff
@@ -0,0 +1,335 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/kong/deck/crud"
+	"github.com/kong/deck/konnect"
+	"github.com/kong/deck/state"
+	"github.com/kong/go-kong/kong"
+)
+
+func eventFromArg(arg crud.Arg) crud.Event {
+	event, ok := arg.(crud.Event)
+	if !ok {
+		panic("unexpected type, expected diff.Event")
+	}
+	return event
+}
```

Comment:
this is a constructor for a type in package `crud` that takes an argument which lives in package `crud` as well.

Probably makes more sense in that package, no?

Reply:
The type used to live in the diff package and was refactored into the crud package as part of this refactor. What you suggest makes sense, will do.